sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/windowExpressions.scala

```diff
@@ -474,6 +479,52 @@ case class Lag(input: Expression, offset: Expression, default: Expression)
   override val direction = Descending
 }
 
+/**
+ * The NthValue function returns the value of `input` at the row that is the `offset`th row of
+ * the window frame (counting from 1). Offsets start at 0, which is the current row. When the
+ * value of `input` is null at the `offset`th row or there is no such an `offset`th row, null
+ * is returned.
+ *
+ * @param input expression to evaluate `offset`th row of the window frame.
+ * @param offset rows to jump ahead in the partition.
+ */
+@ExpressionDescription(
+  usage = """
+    _FUNC_(input[, offset]) - Returns the value of `input` at the row that is the`offset`th row
+      of the window frame (counting from 1). If the value of `input` at the `offset`th row is
+      null, null is returned. If there is no such an offset row (e.g., when the offset is 10,
+      size of the window frame less than 10), null is returned.
```

Comment:
> If there is no such an offset row, null is returned.

Is this behavior consistent with other mainstream DBs?

Reply:
`If there is no such an offset row, null is returned.` means `If the offset greater than the number of values in the window, null is returned.`
`PostgreSQL`,`Vertica`,`Oracle` and `Presto` all clearly stated this requirement.